adagrams/game.py

```diff
@@ -1,11 +1,135 @@
+        letter_grabbed = letters_remaining_pool.pop(random.randint(0,index))
+        letters_grabbed.append(letter_grabbed)
+
+    return(letters_grabbed)
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+    letter_bank_remaining = letter_bank.copy()
+
+    for letter in word:
+        # prevent the tiles that are used being used again ie "DOOR" cannot be made with D,O,R,X,X,X
+        if letter in letter_bank_remaining:
+            letter_bank_remaining.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    
+    score = 0
+    word = word.upper()
+    for letter in word:
+        # score saved as values to the keys(letter)
+        score += VALUES_DICT[letter]
+    if len(word) >= 7:
+        score += 8
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    max_score = 0
+    max_word = ""
+
+    for word in word_list:
+        score = score_word(word)
+        # print(word, score, len(word))
+        
+        # if the score is greater than the current max, make it the new man
+        if score > max_score:
+            max_score = score
+            max_word = word
+        # if the scores are equal, have conditionals for tie breakers
+        elif score == max_score:
+            # if scores are the same but the existing max word lg is 10, or if theyre the same length
+            if len(max_word) == 10 or len(word) == len(max_word):
+                # when they are the same length (and score), then choose the first instance
+                # which would be the current max word
+                # continue skips over this loop to the next word in either cases
+                continue
+            # if scores are the same, but the new word is 10 or longer than the current max word
+            elif len(word) == 10 or len(word) < len(max_word):
+                # the score becomes the max word
+                max_score = score
+                max_word = word
+
+    return max_word, max_score    
+        
+# generated by the function at the end
+# saved as constant list rather than a dict, bc it was an easier data structure
+# the list is unchanged so every function call, it goes back to this
+LIST_OF_LETTERS = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'N', 'N', 'N', 'N', 'N', 'N', 'B', 'B', 
+                    'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'C', 'C', 'P', 'P', 'D', 'D', 'D', 'D', 'Q', 
+                    'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'R', 'R', 'R', 'R', 'R', 
+                    'R', 'F', 'F', 'S', 'S', 'S', 'S', 'G', 'G', 'G', 'T', 'T', 'T', 'T', 'T', 'T', 'H', 
+                    'H', 'U', 'U', 'U', 'U', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'V', 'V', 'J', 
+                    'W', 'W', 'K', 'X', 'L', 'L', 'L', 'L', 'Y', 'Y', 'M', 'M', 'Z']
+
+
+
+VALUES_DICT = {
```

Comment:
Constant variables are generally declared at the top of file after the imports and before your first function.